src/net/server/connection.rs

```diff
@@ -0,0 +1,633 @@
+use crate::base::{Message, StreamTarget};
+use crate::net::server::buf::BufSource;
+use crate::net::server::message::MessageProcessor;
+use crate::net::server::message::{ContextAwareMessage, MsgProvider};
+use crate::net::server::metrics::ServerMetrics;
+use crate::net::server::middleware::chain::MiddlewareChain;
+use crate::net::server::service::{
+    CallResult, Service, ServiceCommand, ServiceError,
+};
+
+use chrono::{DateTime, Utc};
+use core::ops::ControlFlow;
+use core::sync::atomic::Ordering;
+use core::time::Duration;
+use std::io;
+use std::net::SocketAddr;
+use std::sync::Arc;
+use tokio::io::{
+    AsyncRead, AsyncReadExt, AsyncWrite, AsyncWriteExt, ReadHalf, WriteHalf,
+};
+use tokio::sync::mpsc::Sender;
+use tokio::sync::{mpsc, watch};
+use tokio::task::JoinHandle;
+use tracing::{debug, error};
+
+//------------ Connection -----------------------------------------------
+
+pub struct Connection<Stream, Buf, Svc>
```

Comment:
I preferred to have trait bounds that don't cause failure later just because you can instantiate the `struct` but then cannot use it due to requirements on the impl of key functions. I will change it to keep it consistent with the existing vision for `domain` but am still concerned that this leads to a worse developer experience.

FYI the minimum bounds also include `Svc: Service<Buf::Output>` otherwise the reference to `Svc::Target` doesn't work.

Reply:
So, I looked at relaxing trait bound requirements but did it as part of PR #286 in commit 983f153a1a07498b9fa9de9658762b0c20bef65a, as I'd already made some other simplifications there to the `Service` trait (and because I was working on that branch at the time...). I'll resolve this.